org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/SelectionDependentHandler.java

```diff
@@ -28,29 +32,30 @@
 
     @Override
     public void setEnabled(Object evaluationContext) {
-        boolean enabled;
-        TaskView taskView = getTaskView();
-        if (taskView != null) {
-            NodeSelection selectionHistory = taskView.getSelection();
-            enabled = isEnabledFor(selectionHistory);
-        } else {
-            enabled = false;
+        if (evaluationContext instanceof IEvaluationContext) {
+            IEvaluationContext evalContext = (IEvaluationContext) evaluationContext;
+            // getting the current ISelection from the IEvaluationContext
+            // See https://wiki.eclipse.org/Command_Core_Expressions
+            Object variable = evalContext.getVariable(ISources.ACTIVE_CURRENT_SELECTION_NAME);
+            if (variable instanceof ISelection) {
+                setBaseEnabled(isEnabledFor(NodeSelection.from((IStructuredSelection) variable)));
+            }
         }
-        setBaseEnabled(enabled);
     }
 
     protected NodeSelection getSelectionHistory(ExecutionEvent event) {
         TaskView taskView = getTaskView(event);
-        if (taskView == null) {
-            throw new IllegalStateException(String.format("Cannot execute command '%s' in current window.", getCommandName(event)));
-        }
-
-        NodeSelection selectionHistory = taskView.getSelection();
-        if (!isEnabledFor(selectionHistory)) {
-            throw new IllegalStateException(String.format("Cannot execute command '%s' for current selection.", getCommandName(event)));
+        if (taskView != null) {
+            NodeSelection selectionHistory = taskView.getSelection();
+            if (!isEnabledFor(selectionHistory)) {
+                throw new IllegalStateException(
+                        String.format("Cannot execute command '%s' for current selection.", getCommandName(event)));
+            }
+            return selectionHistory;
+        } else {
+            ISelection currentSelection = HandlerUtil.getCurrentSelection(event);
+            return NodeSelection.from(currentSelection);
```

Comment:
The whole method should contain only the two lines above. The `getTaskView()`, `getTaskView(ExecutionEvent)` and the `getCommandName(ExecutionEvent)` is no longer used and can be deleted.